src/ray/core_worker/core_worker.cc

```diff
@@ -2304,7 +2311,7 @@ void CoreWorker::PlasmaCallback(SetResultCallback success,
   SubscribeToPlasmaAdd(object_id);
 
   // Check in-memory store in case object became ready *before* SubscribeToPlasmaAdd.
-  if (Get(std::vector<ObjectID>{object_id}, 0, &vec).ok() && vec.size() > 0) {
+  if (Contains(object_id, &object_is_local).ok() && object_is_local) {
```

Comment:
Hmm now that I'm looking at this code more carefully, isn't there still an issue with this block of code? I get the race condition that this is supposed to guard against, but the worker doesn't wait for the raylet to reply that the subscription completed. So there's still a window after this block of code where the object could appear but we haven't subscribed yet.

It seems like a better approach would be to remove this block completely and just have the raylet fire the callbacks if the object is already local when it receives the subscribe request.

Reply:
Yep, agreed. Another potential solution would be to make the `SubscribeToPlasmaAdd` RPC synchronous and optionally return a boolean indicating that the object is ready. I don't see a significant to that and it adds a codepath, so preference for @stephanie-wang's suggestion.

We could actually even drop the initial `Get` call in that case, though that might be a worthwhile optimization.